src/java/org/apache/cassandra/index/sai/disk/hnsw/CassandraOnHeapHnsw.java

```diff
@@ -81,11 +81,7 @@ public CassandraOnHeapHnsw(AbstractType<?> termComparator, IndexWriterConfig ind
         serializer = (VectorType.VectorSerializer)termComparator.getSerializer();
         vectorValues = new ConcurrentVectorValues(((VectorType)termComparator).dimension);
         similarityFunction = indexWriterConfig.getSimilarityFunction();
-        // We need to be able to inexpensively distinguish different vectors, with a slower path
-        // that identifies vectors that are equal but not the same reference.  A comparison-
-        // based Map (which only needs to look at vector elements until a difference is found)
-        // is thus a better option than hash-based (which has to look at all elements to compute the hash).
-        postingsMap = new ConcurrentSkipListMap<>(Arrays::compare);
+        postingsMap = new ConcurrentHashMap<>();
```

Comment:
`VectorUpdateDeleteTest#updateTest` failed because of this change..

Reply:
```
    public void remove(ByteBuffer term, T key)
    {
        assert term != null && term.remaining() != 0;

        var vector = serializer.deserializeFloatArray(term);
        var postings = postingsMap.get(vector);
```

With CHM, `postingsMap.get(vector)` is not able to find the corresponding `VectorPostings`

Reply:
I think CHM is not using proper `equals` and `hashcode` for `float[]`. We have to use CSLM..